### PR TITLE
Feature/update claim with expected delivery date

### PR DESCRIPTION
**Pull request by dwybourn**

### CODE REVIEW

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/NewClaimService.java [hunk 1]
```diff
@@ -42,25 +44,54 @@
...
+        log.info("Updated claim: {},  fields updated {}", claim.getId(), updatedFields);
+        eventAuditor.auditUpdatedClaim(claim, updatedFields);
+        return claim;
+    }
+
+    private List<String> updateClaim(Claim claim, Claimant claimant) {
+        if (!Objects.equals(claim.getClaimant().getExpectedDeliveryDate(), claimant.getExpectedDeliveryDate())) {
+            claim.getClaimant().setExpectedDeliveryDate(claimant.getExpectedDeliveryDate());
+            claimRepository.save(claim);
+            return singletonList("expectedDeliveryDate");
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/NewClaimService.java]: Can we have an enum for the fieldnames please as they need to be well defined and match up to the UI.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/NewClaimServiceTest.java [hunk 2]
```diff
@@ -170,17 +175,74 @@ void shouldSaveClaimantWithClaimStatus(EligibilityStatus eligibilityStatus) {
...
         //given
-        Claimant claimant = aValidClaimant();
-        given(eligibilityAndEntitlementService.evaluateNewClaimant(any()))
-                .willReturn(EligibilityAndEntitlementDecision.buildWithStatus(DUPLICATE));
+        Claimant existingClaimant = aClaimantWithExpectedDeliveryDate(null);
+        LocalDate expectedDeliveryDate = LocalDate.now().plusMonths(6);
+        Claimant newClaimant = aClaimantWithExpectedDeliveryDate(expectedDeliveryDate);
+        Claim existingClaim = aClaimWithClaimant(existingClaimant);
+        UUID existingClaimId = UUID.randomUUID();
+        given(eligibilityAndEntitlementService.evaluateNewClaimant(any())).willReturn(EligibilityAndEntitlementDecision.builder()
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/NewClaimServiceTest.java]: Can we use a test data factory for the `EligibilityAndEntitlementDecision` here?

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/NewClaimServiceTest.java [hunk 3]
```diff
@@ -170,17 +175,74 @@ void shouldSaveClaimantWithClaimStatus(EligibilityStatus eligibilityStatus) {
...
+        given(claimRepository.findById(any())).willReturn(Optional.of(existingClaim));
 
         //when
-        ClaimResult result = newClaimService.createClaim(claimant);
+        ClaimResult result = newClaimService.createOrUpdateClaim(newClaimant);
 
         //then
-        verify(eligibilityAndEntitlementService).evaluateNewClaimant(claimant);
+        assertThat(result.getClaim()).isEqualTo(existingClaim);
+        assertThat(result.getClaim().getClaimant()).isEqualTo(existingClaimant);
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/NewClaimServiceTest.java]: To make it a bit clearer that this is matching by id, could we just compare the ids here?

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/NewClaimServiceTest.java [hunk 4]
```diff
@@ -170,17 +175,74 @@ void shouldSaveClaimantWithClaimStatus(EligibilityStatus eligibilityStatus) {
...
+        verifyZeroInteractions(messageQueueDAO);
+    }
+
+    @Test
+    void shouldThrowExceptionWhenExistingClaimNotFound() {
+        //given
+        LocalDate expectedDeliveryDate = LocalDate.now().plusMonths(6);
+        Claimant newClaimant = aClaimantWithExpectedDeliveryDate(expectedDeliveryDate);
+        UUID existingClaimId = UUID.randomUUID();
+        given(eligibilityAndEntitlementService.evaluateNewClaimant(any())).willReturn(EligibilityAndEntitlementDecision.builder()
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/NewClaimServiceTest.java]: Test data factory again?

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/NewClaimServiceTest.java (hunk 3 above)
Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/NewClaimServiceTest.java]: Updated to just check the claim and eed which I think is a bit clearer.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/NewClaimService.java (hunk 1 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/NewClaimService.java]: Added enum

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/NewClaimServiceTest.java (hunk 2 above)
Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/NewClaimServiceTest.java]: Sticking with builder due to number of fields being set.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/NewClaimServiceTest.java (hunk 4 above)
Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/NewClaimServiceTest.java]: Sticking with builder due to number of fields being set.

**Pull request merged**